provider/resource_kpi_threshold_template.go

```diff
@@ -375,13 +381,15 @@ func kpiThresholdTemplate(ctx context.Context, tfKpiThresholdTemplate modelKpiTh
 		policy["time_blocks"] = timeBlocks
 		aggregateThresholds, err := kpiThresholdThresholdSettingsAttributesToPayload(tfpolicy.AggregateThresholds)
 		if err != nil {
-			return nil, err
+			diags.AddError("Failed to populate kpi threshold template.", err.Error())
+			return
 		}
 		policy["aggregate_thresholds"] = aggregateThresholds
 
 		entityThresholds, err := kpiThresholdThresholdSettingsAttributesToPayload(tfpolicy.EntityThresholds)
 		if err != nil {
-			return nil, err
+			diags.AddError("Failed to populate kpi threshold template.", err.Error())
```

Comment:
Thank you for refactoring. I was thinking about the right place to start return diags instead of errors because of base returns error. And with this diff it clearly seen that cleaner just to convert errors returned from Base

Reply:
the idea being that Base is not terraform-specific at all, so returns errors, but the resource is the interface, so err -> diag here makes sense to me.